### DISCUSSION TITLE
Liens entre équipements physiques et bâtiment/salle/baie

### DISCUSSION THREAD

**Discussion by fxy-afau** [dbarzin/mercator#243] [Ideas]
Bonjour Didier,

Tout d'abord je tenais à vous remercier de mettre à disposition l'application Mercator qui permet de réaliser de manière simple et élégante la cartographie d'un système d'information.

Sur la plupart des équipements physiques (stockage,périphérique,téléphone,routeur,commutateur,etc), il est possible de choisir un site et un bâtiment et une baie.
La baie étant liée à un bâtiment et le bâtiment à un site, ne pouvons-nous pas dans ces écrans ne permettre la sélection que d'un seul élément parmi site/bâtiment/baie ? Un peu comme cela est fait pour les flux dans la vue applicative.
Les champs seraient donc mutuellement exclusifs, contrainte qu'il serait possible d'implémenter également au niveau de la base de données avec une contrainte de type CHECK, par exemple :
`ALTER TABLE MyTable`
`ADD CONSTRAINT CheckOnlyOneColumnIsNull`
`CHECK`
`(`
`    ( CASE WHEN col1 IS NULL THEN 0 ELSE 1 END`
`    + CASE WHEN col2 IS NULL THEN 0 ELSE 1 END`
`    + CASE WHEN col3 IS NULL THEN 0 ELSE 1 END`
`    ) = 1`
`)`

Cordialement,
Antoine Fau

**Comment by dbarzin** [comment]
Il faut vérifier si il est possible d'implémenter cela dans Laravel car le framework utilisé est indépendant de la base de données.
Note que cela n'apporte pas beaucoup d'avantages du point de vue de la modélisation de la cartographie.
Il ne sera juste plus possible de créer un équipement sans devoir spécifier où il se trouve.

**Comment by fxy-afau** [comment]
Oui en effet, donc ce serait 1 seul non null ou tous null.
Au delà de la contrainte en base de données, est ce que cela ferait du sens d'implémenter cette contrainte sur les écrans des équipements physiques ?

**Comment by dbarzin** [comment]
Cela oblige ces équipements à se trouver dans une baie. Ce qui n'est pas toujours le cas.

**Comment by fxy-afau** [comment]
Non en fait on pourrait sélectionner soit le site, soit le bâtiment, soit la baie ou aucun des trois

**Comment by dbarzin** [comment]
Une option dans les paramètres de l'utilisateur qui permet de sélectionner
la localisation soit directement ou en cascade soit en passant par site -
salle - baie ?

Le jeu. 20 oct. 2022 à 10:54, Antoine Fau ***@***.***> a
écrit :

**Comment by dbarzin** [comment]
Autre idée, lors de la sélection de la localisation avoir la liste de toutes les baies si le site ou la salle ne sont pas encore sélectionnés.
